File: src/main/java/org/shark/example/service/quartz/CreateJobService.java
package org.shark.example.service.quartz;

import lombok.RequiredArgsConstructor;
import org.quartz.*;
import org.shark.example.service.base.pojo.ResponseDto;
import org.shark.example.service.quartz.pojo.CreateJobDto;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor
@Service
public class CreateJobService {

    private final Scheduler scheduler;

    public ResponseDto<Void> start(CreateJobDto createJobDto) {
        JobDetail jobDetail = JobBuilder.newJob(ExampleJob.class).
                withIdentity(createJobDto.getJobName(), createJobDto.getJobGroup()).build();

        Trigger trigger = TriggerBuilder.newTrigger().withIdentity(createJobDto.getJobName(), createJobDto.getJobGroup())
                .startAt(DateBuilder.futureDate(1, DateBuilder.IntervalUnit.SECOND))
                .withSchedule(CronScheduleBuilder.cronSchedule(createJobDto.getCronExpression())).startNow().build();

        try {
            scheduler.scheduleJob(jobDetail, trigger);
            if (!scheduler.isShutdown()) {
                scheduler.start();
            }
        } catch (SchedulerException e) {
            throw new RuntimeException(e);
        }

        return ResponseDto.<Void>builder().status(1).build();
    }
}

File: src/main/kotlin/org/shark/example/service/quartz/CreateJobService.kt
package org.shark.example.service.quartz

import org.quartz.*
import org.shark.example.service.base.pojo.ResponseDto
import org.shark.example.service.quartz.pojo.CreateJobDto
import org.springframework.stereotype.Service

@Service
class CreateJobService(
    private val scheduler: Scheduler
) {
    fun start(createJobDto: CreateJobDto): ResponseDto<Void> {
        val jobDetail = JobBuilder.newJob(ExampleJob::class.java)
            .withIdentity(createJobDto.jobName, createJobDto.jobGroup).build()

        val trigger = TriggerBuilder.newTrigger()
            .withIdentity(createJobDto.jobName, createJobDto.jobGroup)
            .startAt(DateBuilder.futureDate(1, DateBuilder.IntervalUnit.SECOND))
            .withSchedule(CronScheduleBuilder.cronSchedule(createJobDto.cronExpression)).startNow().build()

        try {
            scheduler.scheduleJob(jobDetail, trigger)
            if (!scheduler.isShutdown) {
                scheduler.start()
            }
        } catch (e: SchedulerException) {
            throw RuntimeException(e)
        }

        return ResponseDto.builder<Void>().status(true).build()
    }
}
